zone.go

```diff
@@ -486,20 +494,29 @@ func (api *API) ZoneSetVanityNS(zoneID string, ns []string) (Zone, error) {
 	return zone, nil
 }
 
-// ZoneSetPlan changes the zone plan.
-func (api *API) ZoneSetPlan(zoneID string, plan ZonePlan) (Zone, error) {
-	zoneopts := ZoneOptions{Plan: &plan}
-	zone, err := api.EditZone(zoneID, zoneopts)
+// ZoneSetPlan sets the rate plan of an existing zone.
+//
+// Valid values for `planType` are "CF_FREE", "CF_PRO", "CF_BIZ" and
+// "CF_ENT".
+//
+// API reference: https://api.cloudflare.com/#zone-subscription-update-zone-subscription
+func (api *API) ZoneSetPlan(zoneID string, planType string) error {
+	zonePayload := zoneSubscriptionRatePlanPayload{}
+	zonePayload.RatePlan.ID = planType
+
+	uri := fmt.Sprintf("/zones/%s/subscription", zoneID)
+
+	_, err := api.makeRequest("POST", uri, zonePayload)
```

Comment:
I think subscriptions will require POST on new zone plans, but PUT for updates once a subscription already exists

Reply:
I don't know if this begs the question of whether a subscription is a resource that should be controlled by TF. Seems like overkill, but I'm not sure.

Reply:
> I think subscriptions will require POST on new zone plans, but PUT for updates once a subscription already exists

Gotcha. I might look to also add a method for `ZoneUpdateRatePlan` too. Thoughts?

> I don't know if this begs the question of whether a subscription is a resource that should be controlled by TF. Seems like overkill, but I'm not sure.

It's definitely a fair question and my preference is to have as much as possible available in Terraform where we aren't doing anything that is unmaintainable or difficult to work with and then let the consumer decide how much they put into code. FWIW, we manage all of our infrastructure in a repository of some kind (Terraform, Cloudformation, bash scripts); this just means one less thing we handle outside of Terraform for Cloudflare.

Reply:
Having an update makes sense. You will just need the subscription id for that.

Yeah, if the subscription state is kept in the zone object that might work best for now. If at some point we add full support for subscriptions, then we can maybe move it out.

Reply:
Sweet, have added an update method via 2182e7e